drivers/usb_c/tcpc/ucpd_stm32.c

```diff
@@ -1027,21 +1027,20 @@ static int ucpd_transmit_data(const struct device *dev,
 /**
  * @brief Tests if a received Power Delivery message is pending
  *
- * @retval true if message is pending, else false
+ * @retval 0 if message is pending, else -ENODATA
```

Comment:
Searching for other APIs that return `int`, but have `is` in the name, I found a few examples that return 0 or 1 for non-errors. 
[gpio_pin_is_input](https://github.com/zephyrproject-rtos/zephyr/blob/d885048637d6070182cf3ca832631dc91c85ad14/include/zephyr/drivers/gpio.h#L1080)
[rtc_alarm_is_pending](https://github.com/zephyrproject-rtos/zephyr/blob/d885048637d6070182cf3ca832631dc91c85ad14/include/zephyr/drivers/rtc.h#L344)

I prefer this approach so that negative return values are only used for errors.

Reply:
I was talking with @sambhurst to find a best approach to this function and we found the ENODATA error code so we thought about using it. However, in the next commit this function is merged with the receive data one, so the behavior and return codes are different.
But anyway, thanks for sharing the functions you found, I will remember to use that approach.

Reply:
Thanks Keith, I like this approach too.